woocommerce/woocommerce-gutenberg-products-block
# Reviews by Product: split 'block.js' into smaller chunks

This PR splits `blocks.js` into `editor-block.js` and `frontend-block.js` and extracts the common components to a `/assets/js/frontend/components` folder.

Some benefits from this:
* All files are smaller now, so it should be easier to work with them.
* Having a different block for the editor and the frontend allows us to add dependencies to the editor block (_ie_ `debounce`) whithout afecting the frontend block. It also keeps both blocks simpler: the editor block doesn't need the `appendReviews()` method, for example, while the frontend block doesn't need to handle attribute updates in `componentDidUpdate()`.
* The shared code between both lies in `utils` and `components`, so duplication of code should be small.
* Having extracted the `components` and `utils` code, it should be easier to develop the _Reviews by Category_ block without duplicating a lot of code.
* Creates an `/assets/js/frontend/` folder to group components and HOCs used in the frontend.

### How to test the changes in this Pull Request:

1. Create a post with a _Reviews by Product_ block and interact with it.
2. Verify everything is still working fine both in the editor and the frontend.
